pkg/git/gogit_test.go

```diff
@@ -84,6 +84,18 @@ var _ = Describe("Clone", func() {
 	})
 })
 
+var _ = Describe("ValidateAccess", func() {
+	It("validate access to a given repository successfully", func() {
+		err := gitClient.ValidateAccess(context.Background(), "https://github.com/githubtraining/hellogitworld", "master")
+		Expect(err).ShouldNot(HaveOccurred())
+	})
+
+	It("fails to validate access to a possible private repository", func() {
+		err := gitClient.ValidateAccess(context.Background(), "https://github.com/notexisted/repo", "master")
+		Expect(err).Should(HaveOccurred())
```

Comment:
I did a test to try to see if the Clone method would fail differently on either empty repo or non-existed branch using an invalid ssh key but it always fails with the same error `ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain`. So there is no need to check for those specific errors.